torch/onnx/symbolic_opset9.py

```diff
@@ -1019,8 +1019,8 @@ def batch_norm(g, input, weight, bias, running_mean, running_var, training, mome
     out = g.op("BatchNormalization", input, weight, bias, running_mean, running_var,
                epsilon_f=eps,
                momentum_f=1 - momentum,
-               outputs=1 if not training else 5)
-    if not training:
+               outputs=1 if not sym_help._training_mode else 5)
```

Comment:
For here and below, why are we switching from using `training`?

Reply:
since we have a training parameter now it would be more consistent to use it everywhere.
Do you think it would be better to keep using the input 'training' instead?
Maybe I could add an assertion that they are equal here?

Reply:
I see, I agree we should use the new parameter for consistency. Let's add a helper function asserting that `training` input is equivalent to training parameter.

Reply:
@BowenBao @spandantiwari,

For context: previously we were using the param "training" from the inputs of the symbolics of dropout and bn. After adding the attribute _training_mode, we are now using this parameter instead for consistency.

To address this comment, I added a function in symbolic_helper to assert that the "training" param for bn and dropout has the same value as the new _training_mode attribute.

I saw failures happening in test_operators.py; and after debugging, I realized that with lambdas the "training" param does not behave as expected... In that case the model is a FuncModule() with a ParameterList(), and setting the model's mode does not modify the value of "training" in bn and dropout.

In that case, we would want to trust the _training_mode param rather than the the training param given in the symbolic.
I changed the assert to throw a warning instead, we would then use the value of _training_mode for the model's mode, but wanted to consult you in case you have a better idea of how to deal with this situation.